sdk/include/reactos/DX_Shell.h

```diff
@@ -0,0 +1,228 @@
+/*
+ * PROJECT:     ReactOS Dirty Shell Extension
+ * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
+ * PURPOSE:     Removing length limit of MAX_PATH from shell
+ * COPYRIGHT:   Copyright 2023 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#pragma once
+
+#ifndef __cplusplus
+    #error Use C++ compiler for this header
+#endif
+
+#define DX_PathIsRootW          PathIsRootW
+#define DX_PathIsRelativeW      PathIsRelativeW
+#define DX_PathIsUNCW           PathIsUNCW
+
+inline VOID DX_PathAddBackslashW(CStringW& strPath)
+{
+    INT nLength = strPath.GetLength();
+    if (nLength > 0 && strPath[nLength - 1] == L'\\')
+        return;
+    strPath += L'\\';
+}
+
+inline VOID DX_PathRemoveBackslashW(CStringW& strPath)
+{
+    INT nLength = strPath.GetLength();
+    if (nLength > 0 && strPath[nLength - 1] == L'\\')
+        strPath = strPath.Left(nLength - 1);
+}
+
+inline VOID DX_PathAppend(CStringW& strPath, LPCWSTR pszAppend)
+{
+    DX_PathAddBackslashW(strPath);
+    strPath += pszAppend;
+}
+
+inline VOID DX_PathRemoveFileSpecW(CStringW& strPath)
+{
+    LPWSTR pszPath = strPath.GetBuffer();
+    PathRemoveFileSpecW(pszPath);
+    strPath.ReleaseBuffer();
+}
+
+inline VOID DX_PathAddExtensionW(CStringW& strPath, LPCWSTR pszDotExt)
+{
+    strPath += pszDotExt;
+}
+
+inline VOID DX_PathRemoveExtensionW(CStringW& strPath)
+{
+    LPWSTR pszPath = strPath.GetBuffer();
+    PathRemoveExtensionW(pszPath);
```

Comment:
Same comment as for the PathRemoveFileSpec.

Reply:
`PathCchRemoveExtension` returns `HRESULT`.
But `PathRemoveExtension` returns `VOID`.